### PR TITLE
Use type array for serverProperties and customProperties settings

### PR DESCRIPTION
**Pull request by gabro**
Starting from last July VSCode release, this type of setting finally has a UI 🎉

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -108,9 +108,7 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
     : defaultServerVersion;
   const serverProperties: string[] = workspace
     .getConfiguration("metals")
-    .get<string>("serverProperties")!
-    .split(" ")
-    .filter(e => e.length > 0);
```
Comment (gabro) [inline src/extension.ts]: I've verified that the UI won't let you input a blank string (not even just spaces), so this I've removed this check.

Same thing below

**Comment by tgodzik** [comment]
@gabro I think we need to add some kind of migration - if user had a setting as string, we need to replace it with a lsit

**Comment by gabro** [comment]
@tgodzik you are right, I'll work on a fix for this soon.

**Comment by olafurpg** [comment]
This is awesome :D

**Pull request merged**